### DISCUSSION TITLE
Flow within the same basin in flow.arrow

### DISCUSSION THREAD

**Discussion by Joris-J** [Deltares/Ribasim#1477] [Q&A] [answered]
Hey!

I am reading out the output files of a ribasim model. Currently, I am looking at the 'flow.arrow' file. In this file flows between different nodes can be found. 
My question lies in the fact that I can see flows where the 'from_node_id' is the same as the 'to_node_id'. This can be found for all basins present in my model. How can I interpret this flow?

**Comment by visr** [comment, marked as answer]
Hi. Yes we used to do this, but in recent releases we don't have these anymore, hence its description is also removed from the docs.

The way to interpret this is as a boundary flow into the model at the given node ID. If it is negative, it is a flow out of the model. We initially added these boundary flows because they are needed to make a complete water balance based on the `flow.arrow` file. However, as this question also shows, its meaning was not obvious, and there was also no way to distinguish for instance `precipitation` and `drainage` into the Basin. Therefore those flows should now be taken from [`basin.arrow`](https://deltares.github.io/Ribasim/core/usage.html#basin---basin.arrow), and `flow.arrow` is only for flow between nodes, not onto nodes.
